artemis-prometheus-metrics-plugin-servlet/src/main/java/com/redhat/amq/broker/core/server/metrics/plugins/ArtemisPrometheusMetricsPluginServlet.java

```diff
@@ -48,13 +48,17 @@ public ArtemisPrometheusMetricsPluginServlet() {
 
    @Override
    protected void doGet(final HttpServletRequest req, final HttpServletResponse resp) throws ServletException, IOException {
-      resp.setStatus(HttpServletResponse.SC_OK);
+      ArtemisPrometheusMeterRegistry currentRegistry = registry;
```

Comment:
the local copy is needed to prevent a race-condition between line 53 and 61 isn't it?
but 'registry' can only go from null to non-null.

Reply:
@jbertram has changed this a bit in the branch for 3.x, see https://github.com/rh-messaging/artemis-prometheus-metrics-plugin/commit/1e0166fc9b2977e067d310dcbc0e6491eb452fba

Reply:
thx, I noticed that the branch was pre-3, but did not look at the details.